gunrock/app/lp/lp_app.cu

```diff
@@ -0,0 +1,294 @@
+  VertexT *h_preds = (mark_pred) ? new VertexT[graph.nodes] : NULL;
+
+  // Allocate problem and enactor on GPU, and initialize them
+  ProblemT problem(parameters);
+  EnactorT enactor;
+  GUARD_CU(problem.Init(graph, target));
+  GUARD_CU(enactor.Init(problem, target));
+  cpu_timer.Stop();
+  parameters.Set("preprocess-time", cpu_timer.ElapsedMillis());
+
+  // perform LP
+  VertexT src;
+  // oh so we are not doing num of runs * srcs, instead or num of runs remains the same its just that we use different sources
+  // that makes sense I guess when we have random sources
+  
+  for (int run_num = 0; run_num < num_runs; ++run_num) {
+    src = srcs[run_num % num_srcs];
+    GUARD_CU(problem.Reset(src, target));
+    GUARD_CU(enactor.Reset(src, target));
+    util::PrintMsg("__________________________", !quiet_mode);
+
+    cpu_timer.Start();
+    GUARD_CU(enactor.Enact(src));
+    cpu_timer.Stop();
+    info.CollectSingleRun(cpu_timer.ElapsedMillis());
+
+    util::PrintMsg(
+        "--------------------------\nRun " + std::to_string(run_num) +
+            " elapsed: " + std::to_string(cpu_timer.ElapsedMillis()) +
+            " ms, src = " + std::to_string(src) + ", #iterations = " +
+            std::to_string(enactor.enactor_slices[0].enactor_stats.iteration),
+        !quiet_mode);
+    if (validation == "each") {
+      GUARD_CU(problem.Extract(h_labels, h_preds));
+      SizeT num_errors = app::lp::Validate_Results(
+          parameters, graph, src, h_labels, h_preds,
+          ref_labels == NULL ? NULL : ref_labels[run_num % num_srcs], (VertexT*)NULL,
+          false);
+    }
+  }
+
+  cpu_timer.Start();
+  // Copy out results
+  GUARD_CU(problem.Extract(h_labels, h_preds));
+  if (validation == "last") {
+    SizeT num_errors = app::lp::Validate_Results(
+        parameters, graph, src, h_labels, h_preds,
+        ref_labels == NULL ? NULL : ref_labels[(num_runs - 1) % num_srcs]);
+  }
+
+  // compute running statistics
+  info.ComputeTraversalStats(enactor, h_labels);
+// Display_Memory_Usage(problem);
+#ifdef ENABLE_PERFORMANCE_PROFILING
+  // Display_Performance_Profiling(&enactor);
+#endif
+
+  // Clean up
+  GUARD_CU(enactor.Release(target));
+  GUARD_CU(problem.Release(target));
+  delete[] h_labels;
+  h_labels = NULL;
+  delete[] h_preds;
+  h_preds = NULL;
+  cpu_timer.Stop();
+  total_timer.Stop();
+
+  info.Finalize(cpu_timer.ElapsedMillis(), total_timer.ElapsedMillis());
+  return retval;
+}
+
+}  // namespace lp
+}  // namespace app
+}  // namespace gunrock
+
+/*
+ * @brief Entry of gunrock_lp function
+ * @tparam     GraphT     Type of the graph
+ * @tparam     LabelT     Type of the labels
+ * @param[in]  parameters Excution parameters
+ * @param[in]  graph      Input graph
+ * @param[out] labels     Return shortest hop distance from source per vertex
+ * @param[out] preds      Return predecessors of each vertex
+ * \return     double     Return accumulated elapsed times for all runs
+ */
```

Comment:
Could you make this comment more detailed? This is where the C API will interface with your application, so it's a good place to add some detail about your arguments, their meaning and expected initialized values. This still says `shortest hop distance`.